### ISSUE TITLE
Collecting Performance Counter by ApplicationInsights.config

### ISSUE THREAD

**Issue by eddynaka** [microsoft/ApplicationInsights-dotnet#1497]
I'm collecting performance counter by adding new counters in the ApplicationInsights.config like this:
```xml
<Counters>
  <Add PerformanceCounter="\Network Interface(InterfaceName)\Bytes Total/sec" ReportAs="Bytes Total/Sec" />
  <Add PerformanceCounter="\Network Interface(InterfaceName)\Current Bandwidth" ReportAs="Current Bandwidth" />
  <Add PerformanceCounter="\Network Interface(InterfaceName)\Bytes Received/sec" ReportAs="Bytes Received/sec" />
  <Add PerformanceCounter="\Network Interface(InterfaceName)\Bytes Sent/sec" ReportAs="Bytes Sent/sec" />
  <Add PerformanceCounter="\Network Interface(InterfaceName)\Packets Outbound Errors" ReportAs="Packets Outbound Errors" />
  <Add PerformanceCounter="\Network Interface(InterfaceName)\Packets Received Errors" ReportAs="Packets Received Errors" />
</Counters>
```

Right now, I couldn't find a way to save those counters based on the correct category / name. Tried to add the ReportAs property with the category but it is saving exactly the same as I'm adding in the XML.

Is it possible to save the correct category and name of the counter?

## Version Info 
SDK Version : 2.11
.NET Version : 4.8					
How Application was onboarded with SDK(VisualStudio/StatusMonitor/Azure Extension) : SDK
OS : Windows 10
Hosting Info (IIS/Azure WebApps/ etc) : WPF

**Comment by cijothomas** [comment]
Can you elaborate this statement **"I couldn't find a way to save those counters based on the correct category / name."** ? I am not sure I understand the problem you are reporting.

**Comment by eddynaka** [comment]
Hi cijo,

When I added those line as pointed, it saved in log analytics exactly as
shown in reportAs property.

Each line in performance counters table has category and name. For those
that I added manually, the category = name = reportAs.

What I want to know is: based on a counter that exists (network interface \
bytes total / sec), is it possible to save the correct category and name?

Thanks!

**Comment by eddynaka** [comment]
Hi Cijo,

below an image showing what I'm saying: 
![image](https://user-images.githubusercontent.com/16199012/70702536-05c5f200-1cad-11ea-8fdc-f560dd5f8531.png)

**Comment by SergeyKanzhelev** [comment]
Looks like a bug, So it seems that ReportAs property breaks the category name

Are you able to debug this in VS? Can you copy/paste event JSON that represent this counter from the debug output window?

**Comment by eddynaka** [comment]
@SergeyKanzhelev , below some examples:
```xml
<Add PerformanceCounter="\Network Interface(Intel[R] Dual Band Wireless-AC 8260)\Bytes Total/sec" ReportAs="\Network Interface(Intel[R] Dual Band Wireless-AC 8260)\Bytes Total/Sec" />
```
```json
{
    "name": "Microsoft.ApplicationInsights.Dev.d9b77e207de34d51839537bcb3c56bd6.Metric",
    "time": "2019-12-12T18:50:22.6659494Z",
    "iKey": "d9b77e20-7de3-4d51-8395-37bcb3c56bd6",
    "tags": {
        "ai.cloud.roleInstance": "MININT-B9K1T2N.southamerica.corp.microsoft.com",
        "ai.internal.sdkVersion": "pc:2.11.2-28447"
    },
    "data": {
        "baseType": "MetricData",
        "baseData": {
            "ver": 2,
            "metrics": [
                {
                    "name": "\\Network Interface(Intel[R] Dual Band Wireless-AC 8260)\\Bytes Total/Sec",
                    "kind": "Aggregation",
                    "value": 101588.1953125,
                    "count": 1,
                    "min": 101588.1953125,
                    "max": 101588.1953125,
                    "stdDev": 0
                }
            ],
            "properties": {
                "CounterInstanceName": "Intel[R] Dual Band Wireless-AC 8260",
                "CustomPerfCounter": "true",
                "DeveloperMode": "true"
            }
        }
    }
}
```

```xml
<Add PerformanceCounter="\Network Interface(Intel[R] Dual Band Wireless-AC 8260)\Current Bandwidth" ReportAs="\Network Interface\Current Bandwidth" />
```
```json
{
    "name": "Microsoft.ApplicationInsights.Dev.d9b77e207de34d51839537bcb3c56bd6.Metric",
    "time": "2019-12-12T18:50:22.6692755Z",
    "iKey": "d9b77e20-7de3-4d51-8395-37bcb3c56bd6",
    "tags": {
        "ai.cloud.roleInstance": "MININT-B9K1T2N.southamerica.corp.microsoft.com",
        "ai.internal.sdkVersion": "pc:2.11.2-28447"
    },
    "data": {
        "baseType": "MetricData",
        "baseData": {
            "ver": 2,
            "metrics": [
                {
                    "name": "\\Network Interface\\Current Bandwidth",
                    "kind": "Aggregation",
                    "value": 400000000,
                    "count": 1,
                    "min": 400000000,
                    "max": 400000000,
                    "stdDev": 0
                }
            ],
            "properties": {
                "CounterInstanceName": "Intel[R] Dual Band Wireless-AC 8260",
                "CustomPerfCounter": "true",
                "DeveloperMode": "true"
            }
        }
    }
}
```

Both are saving like the image that I pasted

**Comment by SergeyKanzhelev** [comment]
what Kusto shows for both cases?

**Comment by eddynaka** [comment]
![image](https://user-images.githubusercontent.com/16199012/70740663-5f9fd980-1cf8-11ea-8122-3f0db7878d1f.png)

**Comment by SergeyKanzhelev** [comment]
hey, I don't think screenshot match those events looking at timestamps

**Comment by eddynaka** [comment]
Just to simplify, let me rephrase the problem and expectation:

**Description of the bug**: when you configure a custom counter to be collected (for example, Network Interface\Bytes Total/Sec), the category and name maintain the same value.

**What I was expecting**: adding a counter from Windows, I would like to category = “Network Interface” and name = “Bytes Total/Sec” (for example).

![image](https://user-images.githubusercontent.com/16199012/70746886-7993e900-1d05-11ea-91cc-0cb9f60fd8bc.png)

Image of the problem.

```xml
<Add PerformanceCounter="\Network Interface(Intel[R] Dual Band Wireless-AC 8260)\Bytes Total/sec" ReportAs="\\Network Interface\\Bytes Total/sec" />
```
XML of the configuration.

```json
{
    "name": "Microsoft.ApplicationInsights.Dev.d9b77e207de34d51839537bcb3c56bd6.Metric",
    "time": "2019-12-12T20:30:22.4849449Z",
    "iKey": "d9b77e20-7de3-4d51-8395-37bcb3c56bd6",
    "tags": {
        "ai.cloud.roleInstance": "MININT-B9K1T2N.southamerica.corp.microsoft.com",
        "ai.internal.sdkVersion": "pc:2.11.2-28447"
    },
    "data": {
        "baseType": "MetricData",
        "baseData": {
            "ver": 2,
            "metrics": [
                {
                    "name": "\\\\Network Interface\\\\Bytes Total/sec",
                    "kind": "Aggregation",
                    "value": 26545.935546875,
                    "count": 1,
                    "min": 26545.935546875,
                    "max": 26545.935546875,
                    "stdDev": 0
                }
            ],
            "properties": {
                "CounterInstanceName": "Intel[R] Dual Band Wireless-AC 8260",
                "CustomPerfCounter": "true",
                "DeveloperMode": "true"
            }
        }
    }
}
```
JSON that the Nuget sent.

**Comment by cijothomas** [comment]
This is done at the ingestion backend which parses name of the counter and separates category and name. 
**But this is done only for the default list of counters, not on anything additionally added.**

**Comment by eddynaka** [comment]
Yeah, I imagined that. But, it's a question of mapping more counters, no? At least, the counters that exists in Windows should be mapped correctly (name and category). Or, we should be able to change that by some kind of configuration.

**Comment by cijothomas** [comment]
This could be treated as a bug in ingestion/backend, or a by design thing (this was only intended to work on default list of counters). Either way this is not something fixable now in SDK alone without backend changes.

I can keep this issue open here for anyone else's reference, but I doubt there will be any fixes in this regard.
I'd assume that perf counters (other than default) is a area of LogAnalytics agent, not Application Insights SDK,
https://docs.microsoft.com/en-us/azure/azure-monitor/platform/data-sources-performance-counters

**Comment by eddynaka** [comment]
Hi @cijothomas , I didn't find any recommendation to use azure monitor & application insights. Don't know if that is a best practices since we are monitoring a client machine.

The idea of this issue is to check if we can:
- adjust ingestion/backend to receive a category name or split the data that we send
- adjust the sdk to send the category (if its possible to change the ingestion)
